xmipptomo/protocols/protocol_phantom_tomo.py

```diff
@@ -131,27 +131,68 @@ def _addTomogram(self, index, acq, xT, yT, zT, radius, height, desc):
         # Write the description file
         fnDescr = self._getExtraPath("phantom.descr")
 
+        # Reduce dimensions form center to avoid particles in the border
+        halfHeight = boxSize/2
+        xT = xT - halfHeight
+        yT = yT - halfHeight
+        zT = zT - halfHeight
+        self.info("Valid offset form center, x, y,z: %s, %s, %s" % (xT, yT, zT))
+
+        # Create a list with possible positions to warranty no overlapping
+        xLen = math.floor(xT/boxSize)
+        yLen = math.floor(yT/boxSize)
+        zLen = math.floor(zT/boxSize)
+        self.info("Possible position matrix (x, y, z): %s, %s, %s." % (xLen, yLen, zLen))
+
+        positions = []
+
+        # Calculate halves of the dimensions
+        xHalf = int(xT / 2)
+        yHalf = int(yT / 2)
+        zHalf = int(zT / 2)
+
+        for z in range(0,zLen):
+            for y in range(0, yLen):
+                for x in range(0, xLen):
+
+                    # Convert positions to actual coordinates with the origin in the center
+                    xPos = int((x * boxSize) + halfHeight) - xHalf
+                    yPos = int((y * boxSize) + halfHeight) - yHalf
+                    zPos = int((z * boxSize) + halfHeight) - zHalf
+                    positions.append((xPos,yPos,zPos))
+
         with open(fnDescr, 'w') as fhDescr:
 
             # For each particle
             for i in range(self.nparticles.get()):
 
+                if len(positions) == 0:
+                    self.warning("There is no more space to add subtomograms without overlapping")
```

Comment:
If this happens, is it possible to report the number of subtomos that have been possible to allocate in the tomogram?

Reply:
True, I'll report the right value. Good catch!